fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBDatabase.java

```diff
@@ -748,10 +748,13 @@ public Transaction createTransaction(Executor executor, @Nullable Map<String, St
      * @return newly created transaction
      */
     private Transaction createTransaction(Executor executor, @Nullable StoreTimer storeTimer, @Nullable Map<String, String> mdcContext, boolean transactionIsTraced) {
+        final boolean enableAssertions = factory.areAssertionsEnabled();
         Transaction transaction = database.createTransaction(executor);
 
         if (storeTimer != null) {
-            transaction = new InstrumentedTransaction(storeTimer, transaction);
+            transaction = new InstrumentedTransaction(storeTimer, transaction, enableAssertions);
+        } else if (enableAssertions) {
+            transaction = new InstrumentedTransaction(new FDBStoreTimer(), transaction, enableAssertions);
```

Comment:
A new `FDBStoreTimer` is a little bit of a weird choice if `StoreTimer` is null, rather than sending `null` down to the `InstrumentedTransaction` (and, um, I guess not instrumenting it). Maybe it's preferable to introducing possible NPEs from that class, though? Not sure.

Reply:
Yeah, it just bothers me that we have this log everywhere of `if (storeTimer != null)`.  I took at stab at creating a `NullStoreTimer` implementation so we can enforce that it is never nullable, but it gets into problems because it doesn't really extend into `NullFDBStoreTimer`. Anyway, I updated this to allow for null :(